### DISCUSSION TITLE
Bed Level adjustment

### DISCUSSION THREAD

**Discussion by Giterdun2** [xblax/flashforge_ad5m_klipper_mod#244] [Q&A] [answered]
Hey all,
I was wondering if someone could explain how to make adjustments to the middle parts of the bed plate. There are adjustment screws on the 4 corners, but none in the middle that I can see.. Whenever I do a bed level, the center part of the bed is always higher than the corners. I have adjusted the corners, but the center is still always higher..  In the attached pic,, I just did a bed level, but I haven't done any corner adjustments in a while and am also using a different plate than before.. 
![bedlevel](https://github.com/user-attachments/assets/5ed0b79c-9877-4ba0-8ec8-b8adfbef6f90)

**Comment by jeffalanmcclain** [comment]
You can't.  And adjusting the corners just helps with z-tilt, not really the bed calibration...which is all sort of one big combined thing on this flash forge, since you don't have separate z-motors...

**Reply by Giterdun2** [reply]
ok thanks.. Just seems kinda strange that it can't be adjusted.. I have adjusted the corners to make them very close to zero after bed levelling, but the center  of the bed is still always higher. If I try to print anything that uses the whole bed, the left side always ends up very rough, and it looks like crap...  oh well... ;)

**Reply by jeffalanmcclain** [reply, marked as answer]
Hum...so, *that* is precisely what bed leveling is for.  Warping of the bed plate at different temps.  You can *never* mechanically trim those out, so you typically get it z-tilt leveled (such that a flat mesh projection has it as level as you can, which the flat mesh projection shows it is) and then the individual areas higher or lower are what the bed mesh does for you.  But, that *should* make it so a first layer print (check out Ellis' Print calibration guide for proper first layer squish) is very nice and uniform.  If it doesn't, I would argue you don't have a good bed mesh.  Or, if you are seeing "rough" print on the left side on layers ABOVE the first layer, this is NOT bed mesh problem...it is probably more too much cooling from the left side bed/chamber fan running too high.

Finally, I will comment that the FlashForge macro for bed mesh is pure crap.  I've revised mine (you can see my submission for a replacement that fixes at least TWO major problems with how it is run in the Pull Requests).  Specifically, it waits for the extruder to cool down to 120'C before starting the bed mesh.  This should normally be done as close to print temps as possible, ideally, however, because PEI can melt and be damaged by hot extruder tips, the normal agreed convention is to run it at a constant 150'C.  Unfortunately, it gets even worse than that, as they LEAVE THE HOT END driver off the entire time it is running bed mesh, which by the time it finishes can be as low as 80'C...imparting OBVIOUS skew to the measurement.  Anyway, give my revised bedmesh macros routine a look / try.  Check out Ellis' guide: https://ellis3dp.com/Print-Tuning-Guide/.  I don't see anything wrong with your bed mesh that would require mechanical adjustments to.

Hope this helps!
Jeff

**Reply by Giterdun2** [reply]
ok thanks Jeff for the explanation.. I'm just somewhat curious as to why it would look like that and not be a bit flatter.. ;) When I print something that is fairly large as I mentioned, the first few layers do print just fine, but as it builds higher up, that's when it starts getting rough, and it seems to be only on the left side, and towards the rear. It also gets noisy when printing over that area when it's rough.. I never considered the cooling fan, but that makes sense as it's closest to that area..  I will check that next time..  But large objects are a rarity anyways.. most things I print take only a small portion of the bed.. I strictly use Klippermod for levelling tho, and I level anytime I change plates or nozzles.. I also haven't updated the A5M firmware for a while due to all the issues that have been reported, and it generally works fine..  thanks again..  Just one question regarding the macros.. It says for auto bed level.. it that when you want to automatically level before a print every time, or does it work when I select Auto Bed Level in the Klippermod menu? Thanks again..

**Reply by jeffalanmcclain** [reply]
Either one on the auto bed level question.  The AUTO bed level just calls the "bed level" routine with specific temps.  That same routine is called when you print and have NOT already loaded a bed mesh (the little tune icon) or you can force it to run.  Either way, it is not the best practice the way they [flashforge] have it running with extruder temp off.  There is also a bug where you could be running a HOTTER bed temp and the routine doesn't check to see if the bed is actually HOTTER than the target bed mesh temp, which then will continue cooling as you run through the calibration, so I fixed that too, with proper checks.

Ok, so your "rough" print is NOT first layer, so that absolutely discounts it being related to the bed mesh / level.  Just curious, but what filament are you printing with and what temps?  I assume you have AUX fan (or the chamber fan) set to be on?  Is it "rough" as in warped on over hangs or look melted or does it look like it just isn't "flowing" hot enough?  All of these I would STRONGLY encourage you to work through the temp tower prints and the flow rate tuning guides on Ellis' page.

Also, take a look at this page to help identify what you mean by "rough" and looks like crap.  It can help zero in on what the issue is.  https://www.simplify3d.com/resources/print-quality-troubleshooting/

**Reply by jeffalanmcclain** [reply]
P.S. it is *possible* that you are not running hot enough on bed temp and getting poor adhesion (again, what filament and temps are you using for extruder/bed and first layers), and the part is curling or warping off the bed on the left side and causing later print layers to look bad?  ABS is notorious for warping and will pull itself off the bed as you progress through the print.

**Reply by Giterdun2** [reply]
hmm.. I didn't see anything in the link that resembled my issue.. Here is a pic of a previous project to give you an idea.. This is Overture Matte Black PLA. I print it at 220, and bed at 60.. I have profiles for various filaments with their recommended settings.. As you can see most of it is fairly smooth, but the left side it pretty rough.. I print mainly with PLA, HS PLA, Silk PLA etc, and some TPU, ABS. I set the temps to roughly what it says on the rolls, and I've successfully printed a lot of things I've designed. ;) I don't generally have an issue with adhesion.. I also use various types of plates like PEI, PC, and some others I've purchased.  I understand that ABS uses higher temps, and I have printed some things without issue.. This only seems to occur with larger objects.. 
![20240818_131639](https://github.com/user-attachments/assets/8c3b3014-a475-4700-82c7-7d56affdd50d)

**Reply by jeffalanmcclain** [reply]
Hum...that DEFINITELY looks related to the chamber fan blowing pattern.  I would say you are possibly over extruding but I would be MORE worried that the part is actually lifting over there and making it "look" like it is over extruding, but the part is just curling up.  This is pretty classic on large / thin prints to peel up from stress warping, though PLA is not normally that bad for that sort of thing (if you said this was ABS or ASA, I would be a lot more certain that was it).  60'C bed temp feels just a *little* high to me for PLA (even HS) and I would try lowering that to around 50'C.  I keep coming back to that looks VERY MUCH like the blower pattern and where it is rough/bad looks like where it is cooling the least, meaning something is running too hot in those areas.  It is *also* possible that you are not getting enough "squish" from your first layer, which is allowing the part to lift on that side/curl up, which means you would need to lower your Z-offset by about 0.1mm or so to get a better first layer adhesion.  I'm assuming you are running your bed mesh (auto bed mesh/calibration) at your target bed temps, as the profile can DRAMATICALLY change over temp.

Here is a response from someone that the TOP LAYER looked VERY much like yours, but it was looking like over extrusion, but ended up being a curled print not having good bed adhesion (again, first layer squish not probably correct).  Not sure if you have chamber fan disabled during first layer, to help with adhesion...either way it is an interaction of the chamber fan (from that pattern) and extrusion multiple and/or first layer adhesion.
https://forum.prusa3d.com/forum/original-prusa-i3-mk3s-mk3-how-do-i-print-this-printing-help/problems-with-over-extruding-top-layer/

It would be worth doing some of the Orca slicer calibrations (temp tower and flow rate test by clicking the Calibration in the top of orca slicer) and Ellis' first layer squish...

**Reply by Giterdun2** [reply]
Ok Thanks.. I'll play with the z offset and fan, but as mentioned it's only when printing larger objects like shown.... I'm not sure I understand how it could be over extruding if most of the print area is just fine...  This was printed a while age, but I don't recall any lifting when it printed.. The PLA filament says 50-70 bed temp.. hence why I usually set it at 60.. I also print pretty much all my PLA filaments at 60 bed temp, and don't seem to have issues.  I do use the auto bed mesh in the filament profiles, and the fan is usually disabled for the first layer..  As I print mostly smaller stuff and this is not an issue for those, I'm not sure when I'll get to play with that again ;) , so thanks for all the information and suggestions.. It's very much appreciated and will certainly be referred to when needed.. ;)
